src/pages/Rockets.jsx

```diff
@@ -1,9 +1,55 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchRockets, reserveRocketAction } from './RocketSlice';
+import './rockets.css';
 
-const Rockets = () => (
-  <>
-    <h1>Rocket page</h1>
-  </>
-);
+const Rockets = () => {
+  const dispatch = useDispatch();
+  const rockets = useSelector((state) => state.rockets.rockets);
+  const status = useSelector((state) => state.rockets.status);
+  const error = useSelector((state) => state.rockets.error);
+
+  useEffect(() => {
+    dispatch(fetchRockets());
+  }, [dispatch]);
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
+  }
+
+  if (status === 'failed') {
+    return <div>{error}</div>;
+  }
+
+  return (
+    <>
+      <div className="rocket-cards">
+        {rockets.map((rocket) => (
+          <div key={rocket.id} className="card-item">
+            <div className="image-container">
+              <img className="rocket-image" src={rocket.flickr_images} alt={rocket.rocket_name} />
+            </div>
+            <div className="information-container">
+              <h2>{rocket.name}</h2>
+              <p>
+                {rocket.reserved && <span className="badge">Reserved</span>}
+                {rocket.description}
+              </p>
+              <button
+                aria-label="reserve-rocket"
+                type="button"
+                value="Reserve Rocket"
+                className={`reserve-rocket${rocket.reserved ? 'reserved' : 'cancelled'}`}
+                onClick={() => dispatch(reserveRocketAction({ id: rocket.id }))}
+              >
+                {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
+              </button>
+            </div>
+          </div>
+        ))}
+      </div>
+    </>
```

Comment:
you can refactor your code better below to take advantage of the bootstrap library as shown below:

```
<>
      <div className="rocket-cards">
        {rockets.map((rocket) => (
          <div key={rocket.id} className="card-item">
            <div className="image-container">
              <img
                className="rocket-image"
                src={rocket.flickr_images}
                alt={rocket.rocket_name}
              />
            </div>
            <div className="information-container">
              <h2>{rocket.name}</h2>
              <p>
                {rocket.reserved && <Badge bg="success">Reserved</Badge>}
                {rocket.description}
              </p>
              {rocket.reserved ? (
                <Button variant="primary" type="button" onClick={() => dispatch(reserveRocketAction(rocket.id))}>Reserve Rocket</Button>
              ) : (
                <Button variant="outline-secondary" type="button" onClick={() => dispatch(reserveRocketAction(rocket.id))}>Cancel Reservation</Button>
              )}
            </div>
          </div>
        ))}
      </div>
    </>
```

![image](https://user-images.githubusercontent.com/22054464/233499994-7273f453-0a8f-45a3-8de6-72d55919d617.png)

![image](https://user-images.githubusercontent.com/22054464/233500028-a78b4f82-4b63-445e-b905-cafbd89e1965.png)